### PR TITLE
K8s nodeos builder

### PR DESCRIPTION
**Pull request by kenan435**

### CODE REVIEW

**Comment by zreigz** [comment]
First squash all commits for two: implementation and documentations

File: README.md [hunk 1]
```diff
@@ -94,26 +43,55 @@ export OS_PASSWORD=secretsecret
...
+After adding the enviroment variables to the env-vars-vm.sh file we need to source the file:
 ```
-. openrc-default.sh
+$ . env-vars-vm.sh 
+	or 
+$ source env-vars-vm.sh
 ```
 
+#### Edit install_nodeos.json with correct parameters 
+Open install.json file and populate the marked fields: name, source_image and networks.
```
Comment (zreigz) [inline README.md]: I don't see file install.json in this project anymore

File: env-vars-cloud.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+export OS_AUTH_URL=https://identity1.citycloud.com:5000/v3
+export OS_IDENTITY_API_VERSION=3
+export OS_DOMAIN_ID=d3c2f05afdbb405ea0b81f589ff2c9e8
+export OS_PROJECT_DOMAIN_ID=d3c2f05afdbb405ea0b81f589ff2c9e8
+export OS_USER_DOMAIN_ID=d3c2f05afdbb405ea0b81f589ff2c9e8
+export OS_TENANT_ID=3eeb47a1bd5a409ebf3ce42b78b1b193
+export OS_PROJECT_ID=3eeb47a1bd5a409ebf3ce42b78b1b193
+export OS_USERNAME=kenan
```
Comment (zreigz) [inline env-vars-cloud.sh]: remove user and password :)

**Comment by zreigz** [comment]
Next I will test it with entire system

**Comment by zreigz** [comment]
PTAL

File: README.md [hunk 3]
```diff
@@ -241,6 +133,68 @@ Build 'openstack' finished.
 
 ```
 
-Now image is available in OpenStack with name centos7-docker-full
+- Image should now be uploaded to OpenStack. You should see it on the HorizonUI under Project -> Images under the name assigned in the json file above.
+
+
+Cloud Example using Openstack on CityCloud
+--------------
+We are assuming that CityCloud or which ever public cloud provider you choose lets you create a reference image. Openstack api requires you to have a reference image on which the further provisioning will be based. After you have successfully created a reference image on your cloud, in this case a CentOS image [CentOS Generic Cloud] (http://cloud.centos.org/centos/7/images/), we will be using packer to further customize that image with necessary packages to bake our own nodeos image.
```
Comment (cheld) [inline README.md]: ### Upload reference image via glance

we should use same chapters for DevStack and OpenStack

File: install_cloud.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      "flavor": "8e00bc89-f2b0-40d3-96d4-05bd67f6e6bf",
+      "use_floating_ip": "true",
+      "floating_ip": "37.153.138.59"
+    }
+  ],
+   "provisioners":[
+     {
+      	"type": "shell",
+      	"execute_command": "{{.Vars}} /bin/sh -x '{{.Path}}'",
+        "script":"scripts/cloud/install-docker.sh"
```
Comment (cheld) [inline install_cloud.json]: I suggest to package all installation steps (docker, docker-bootstrap, nettools) in one file (prepare-docker.sh). This way the installation can be executed with:

 curl -fsSL https://github.com/..../prepare-docker.sh | sh 
curl -fsSL https://github.com/..../master.sh | sh 

on any CentOS installation. Just want to keep all nicely decoupled.

Also, package the minimizing scripts in one file minimize-cloud-image.sh

Comment (cheld) [inline install_cloud.json]: It will also reduce the duplication in this script

**Comment by cheld** [comment]
A few small comments. PTAL

**Comment by cheld** [comment]
Please squash the commits

**Comment by cheld** [comment]
Ping?

File: README.md [hunk 5]
```diff
@@ -1,90 +1,39 @@
-# custom-image-builder
+k8-nodeOS-builder
+==================
 
-## Prepere Openstack
+Prerequisites
+-------------
+- Install [Packer.io](https://packer.io/downloads.html), see [installation instructions](https://packer.io/docs/installation.html).
+- Install [ git ] (https://git-scm.com/downloads)
```
Comment (cheld) [inline README.md]: remove git download

**Comment by cheld** [comment]
Please update and we make another round of review and refinements.

File: env-vars-cloud.sh (hunk 2 above)
Comment (kenan435) [inline env-vars-cloud.sh]: done.

**Comment by kenan435** [comment]
Further enhancements and restructuring of the nodeos builder project.

**Comment by cheld** [comment]
Rename:
OS/CentOS/prepare-docker.sh -> OS/CentOS/setup.sh 

Merge:
OS/CentOS/additional-pkg.sh with OS/CentOS/setup.sh

Remove:
Openstack/DevStack/ to Openstack/ 

Move:
Readme.md to Openstack/Readme.md

New:
Readme.md on top level

Squash

**Comment by cheld** [comment]
Openstack documentation still not clean and professional enough to be put to the public. Please improve

File: README.md [hunk 6]
```diff
@@ -1,90 +1,53 @@
...
-In file install.json you have to enter private network ID.
-Private network should be connected via router to public network.
 
+#### Kubernetes cluster with main and bootstrap docker daemons
+![Alt text](https://raw.githubusercontent.com/kenan435/k8s-nodeos-builder/k8s-nodeos-builder/k8s-docker.png "k8s Docker Based Provisioning")
 
 
-## Install Packer
-Packer is a tool developed by HashiCorp (the people behind Vagrant) to help you create identical cloud images for a variety of different environments. It also allows you to create image templates that are easy to version control and understand what happens during the image creation process.
+### FAQ
```
Comment (cheld) [inline README.md]: I would suggest to move the FAQ to openstack/faq.md, since it is only openstack related

**Comment by cheld** [comment]
One small comment. Other than that it looks fine.

PTAL

**Comment by zreigz** [comment]
LGTM

**Comment by cheld** [comment]
LGTM, but has merge conflicts.

**Pull request merged**